components/support/rs_client/src/http_client.rs

```diff
@@ -22,7 +23,64 @@ pub struct Client {
     pub(crate) base_url: Url,
     pub(crate) bucket_name: String,
     pub(crate) collection_name: String,
-    pub(crate) remote_state: Cell<RemoteState>,
+    pub(crate) remote_state: Mutex<RemoteState>,
+    attachments_base_url: Option<Url>,
+}
+
+/// Data structure representing the top-level response from the Remote Settings.
+/// [last_modified] will be extracted from the etag header of the response.
+pub struct RemoteSettingsResponse {
+    pub records: Vec<Record>,
+    pub last_modified: u64,
+}
+
+#[derive(Deserialize)]
+struct RemoteSettingsRecordsResponse {
+    data: Vec<Record>,
+}
+
+/// A parsed Remote Settings record. Records can contain arbitrary fields, so clients
+/// are required to further extract expected values from the [fields] member.
```

Comment:
The way you're exposing these additional fields to Kotlin with `RsJsonObject` is really slick! 🥇